### ISSUE TITLE
ctrl-r not working in tmux

### ISSUE THREAD

**Issue by zg** [babun/babun#527]
I'm trying to do reverse search with ctrl-r, but it's not working when I'm in `tmux`.

**Comment by asciireign** [comment]
what does happen when you press it?
![ctrlr](https://cloud.githubusercontent.com/assets/12019241/11783160/e3b7d814-a275-11e5-93e7-1d2345ce234f.png)

**Comment by zg** [comment]
Nothing.

**Comment by Serubin** [comment]
Interestingly, I'm having a similar issue on Ubuntu with just ZSH. It may be a tmux+zsh issue.

**Comment by Pistos** [comment]
I was able to work around the problem by explicitly adding this to my `~/.zshrc`:

```
bindkey '^R' history-incremental-search-backward
```

**Comment by tekknolagi** [comment]
Thanks @Pistos ! Helped a lot.

**Comment by tekknolagi** [comment]
...and I'm on macOS using ZSH+Tmux...

**Comment by dmbhatti** [comment]
same issue. same env as @tekknolagi 
noticed after session restore. using resurrect and continuum

**Comment by dmbhatti** [comment]
another fix is to run ```bindkey -e```. That will also fix other ctrl binds not working properly

**Comment by Arnie97** [comment]
TL;DR: `$EDITOR` / `$VISUAL` has rarely known side effects; `unset` them or run `bindkey -e`

If anyone here happens to use (Neo)vi(m) too, then they probably has fallen into the same pitfall as me. In `.zshrc` or [somewhere else in their shell startup scripts](https://blog.flowblok.id.au/2013-02/shell-startup-scripts.html) was `export EDITOR=vim` defined, which does not affect the ZLE behaviour of the current shell, but puts all the nested shells (including tmux sessions) into "vi insert mode":

```c
$ man 1 zshzle

KEYMAPS
       Initially, there are eight keymaps:

       emacs  EMACS emulation
       viins  vi emulation - insert mode

       ...

       In addition to these names, either `emacs' or `viins' is also linked to
       the name `main'.  If one of the VISUAL or EDITOR environment  variables
       contain  the  string `vi' when the shell starts up then it will be `vi‐
       ins', otherwise it will be `emacs'.  bindkey's -e and -v  options  pro‐
       vide a convenient way to override this default choice.

       When  the  editor starts up, it will select the `main' keymap.

       ...
```

Sorry to be off-topic, but this babun thread already ranked 1st on Google, and hence collected `macOS` and `Ubuntu Linux` keywords... Hope my findings help.
